src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -246,57 +289,96 @@
      * Do not use certain named routes.
      * The paramter format is: feedId_routeId,feedId_routeId,feedId_routeId
      * This parameter format is completely nonstandard and should be revised for the 2.0 API, see issue #1671.
+     *
+     * TODO: Needs to be implemented
      */
+    @Deprecated
     public RouteMatcher bannedRoutes = RouteMatcher.emptyMatcher();
 
-    /** Only use certain named routes */
+    /** Only use certain named routes
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     public RouteMatcher whiteListedRoutes = RouteMatcher.emptyMatcher();
 
-    /** Do not use certain named agencies */
+    /** Do not use certain named agencies
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     public HashSet<String> bannedAgencies = new HashSet<String>();
 
-    /** Only use certain named agencies */
+    /** Only use certain named agencies
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     public HashSet<String> whiteListedAgencies = new HashSet<String>();
 
-    /** Do not use certain trips */
+    /** Do not use certain trips
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     public HashMap<FeedScopedId, BannedStopSet> bannedTrips = new HashMap<FeedScopedId, BannedStopSet>();
 
+    /** Set of preferred routes by user.
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     /** Set of preferred routes by user. */
     public RouteMatcher preferredRoutes = RouteMatcher.emptyMatcher();
     
-    /** Set of preferred agencies by user. */
+    /** Set of preferred agencies by user.
+     *
+     * TODO: Needs to be implemented
+     * */
+    @Deprecated
     public HashSet<String> preferredAgencies = new HashSet<String>();
 
     /**
      * Penalty added for using every route that is not preferred if user set any route as preferred. We return number of seconds that we are willing
      * to wait for preferred route.
+     *
+     * TODO Needs to be implemented
      */
+    @Deprecated
     public int otherThanPreferredRoutesPenalty = 300;
 
     /** Set of unpreferred routes for given user. */
+    @Deprecated
```

Comment:
Why are these unpreferred* parameters deprecated without TODO to implement them? Does implementing them differ from a technical point of view from implementing the preferred* parameters or is it just that they are not in use?

Reply:
This PR needs more JavaDoc, all @Deprecated fields need to be explained why they are deprecated.